docs/designs/supporting_pytorch.md

```diff
@@ -0,0 +1,358 @@
+# Design doc about Supporting PyTorch
+
+## introduction
+
+ElasticDL is an open-source distributed deep learning programming framework based
+on TensorFlow 2.x and Kubernetes. At the Google Developer Day event in the fall of
+2019, the ElasticDL team from Ant Financial showed off the first open-source version
+of ElasticDL.The primary design intent of ElasticDL is to simplify distributed programming.
+It allows users to provide only models described with TensorFlow 2.0 API, without
+requiring users to write distributed training process code. As long as the user's
+model definition can be adjusted locally, the model can be trained with large-scale
+ data in a distributed environment, thereby improving R&D efficiency.
+
+However, this project only supports TensorFlow. Considering that PyTorch is more
+widely used in academia, this project will support PyTorch.
+
+## DataLoader
+
+PyTorch has an abstract Dataset class. A Dataset can be anything that has a `__len__`
```

Comment:
We need to consider how to work with [the dynamic task dispatch mechanism](https://github.com/sql-machine-learning/elasticdl/blob/develop/docs/designs/dynamic_data_sharding.md) of ElasticDL.

1. A worker get a task from the master. 
2. A worker reads real data according to the offset in the task

We should consider how to make DataLoader works with task. Maybe we could create a Dataset for each task.

Reply:
Agree with @QiJune , we can link to PyTorch DataLoad, and force on how to make it works with ElasticDL master.